projects/laji/src/app/+project-form/results/common/result-map/result-map.component.ts

```diff
@@ -0,0 +1,306 @@
+import { LajiMapVisualization } from 'projects/laji/src/app/shared-modules/legend/laji-map-visualization';
+import { WarehouseApi } from 'projects/laji/src/app/shared/api/WarehouseApi';
+import { WarehouseQueryInterface } from 'projects/laji/src/app/shared/model/WarehouseQueryInterface';
+import { toHtmlSelectElement } from 'projects/laji/src/app/shared/service/html-element.service';
+import { BehaviorSubject, Observable, combineLatest } from 'rxjs';
+import { map, switchMap, tap } from 'rxjs/operators';
+
+type GatheringCountType = 'ONE' | 'FIVE' | 'TEN' | 'FIFTY' | 'HUNDRED' | 'FIVE_HUNDRED';
+type ObservationProbabilityType = 'ZERO' | 'OVER_ZERO' | 'OVER_TWENTY' | 'OVER_FORTY' | 'OVER_SIXTY' | 'OVER_EIGHTY' | 'HUNDRED';
+
+type ResultVisualizationMode = 'gatheringCount' | 'observationProbability';
+
+type CountMap = Map<string, {lat: string; lon: string; count: number}>;
+type RatioMap = Map<string, {lat: string; lon: string; ratio: number}>;
+type RatioArray = { lat: string; lon: string; ratio: number }[];
+
+interface QueryResult {
+  results: {
+    aggregateBy: {
+      'gathering.conversions.ykj10kmCenter.lat': string;
+      'gathering.conversions.ykj10kmCenter.lon': string;
+    };
+    gatheringCount: number;
+  }[];
+}
+
+const gatheringCountToVisCategory: Record<GatheringCountType, { color: string; label: string }> = {
+  ONE: {
+    color: '#EE82EE',
+    label: '1-4'
+  },
+  FIVE: {
+    color: '#1E90FF',
+    label: '5-9'
+  },
+  TEN: {
+    color: '#00FF00',
+    label: '10-49'
+  },
+  FIFTY: {
+    color: '#FFFF00',
+    label: '50-99'
+  },
+  HUNDRED: {
+    color: '#FFA500',
+    label: '100-499'
+  },
+  FIVE_HUNDRED: {
+    color: '#DC143C',
+    label: '500'
+  },
+};
+
+const observationProbabilityToVisCategory: Record<ObservationProbabilityType, { color: string; label: string }> = {
+  ZERO: {
+    color: '#E6E6E6',
+    label: '0%'
+  },
+  OVER_ZERO: {
+    color: '#EE82EE',
+    label: '0-19%'
+  },
+  OVER_TWENTY: {
+    color: '#1E90FF',
+    label: '20-39%'
+  },
+  OVER_FORTY: {
+    color: '#00FF00',
+    label: '40-59%'
+  },
+  OVER_SIXTY: {
+    color: '#FFFF00',
+    label: '60-79%'
+  },
+  OVER_EIGHTY: {
+    color: '#FFA500',
+    label: '80-99%'
+  },
+  HUNDRED: {
+    color: '#DC143C',
+    label: '100%'
+  },
+};
+
+@Component({
+  selector: 'laji-result-map',
+  templateUrl: './result-map.component.html',
+  styleUrls: ['./result-map.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ResultMapComponent implements OnInit {
+  readonly collections$ = new BehaviorSubject<string[]>([]);
+  readonly taxon$ = new BehaviorSubject<string | undefined>(undefined);
+  @Input() set collections(v: string[]) { this.collections$.next(v); };
+  @Input() set taxon(v: string | undefined) { this.taxon$.next(v); };
+  @Input() taxonOptions$: Observable<{ label: string; value: string }[]>;
```

Comment:
This doesn't need to be an observable, it's simpler if it it's just a normal input. The components using it can use it like `[taxonOptions] = taxonOptions$ | async`